google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java

```diff
@@ -75,12 +179,52 @@ public void resultSetIteration() {
     assertThat(rs.getString(0)).isEqualTo("x");
     assertThat(rs.getLong(1)).isEqualTo(2L);
     assertThat(rs.getBoolean(2)).isTrue();
+    assertThat(rs.getBoolean("f3")).isTrue();
+    assertThat(rs.getDouble("d")).isWithin(0.0).of(dval);
+    assertThat(rs.getDouble(3)).isWithin(0.0).of(dval);
+    assertThat(rs.getString(4)).isEqualTo(s);
+    assertThat(rs.getString("s")).isEqualTo(s);
+    assertThat(rs.getBytes(5)).isEqualTo(ByteArray.copyFrom(byt));
+    assertThat(rs.getBytes("byt")).isEqualTo(ByteArray.copyFrom(byt));
+    assertThat(rs.getTimestamp(6)).isEqualTo(Timestamp.ofTimeMicroseconds(us));
+    assertThat(rs.getTimestamp("ts")).isEqualTo(Timestamp.ofTimeMicroseconds(us));
+    assertThat(rs.getDate(7)).isEqualTo(Date.fromYearMonthDay(yr, month, day));
+    assertThat(rs.getDate("dt")).isEqualTo(Date.fromYearMonthDay(yr, month, day));
+    assertThat(rs.getBooleanArray(8)).isEqualTo(ba);
+    assertThat(rs.getBooleanArray("ba")).isEqualTo(ba);
+    assertThat(rs.getBooleanList(8)).isEqualTo(Booleans.asList(ba));
+    assertThat(rs.getBooleanList("ba")).isEqualTo(Booleans.asList(ba));
+    assertThat(rs.getLongArray(9)).isEqualTo(la);
+    assertThat(rs.getLongArray("la")).isEqualTo(la);
+    assertThat(rs.getLongList(9)).isEqualTo(Longs.asList(la));
+    assertThat(rs.getLongList("la")).isEqualTo(Longs.asList(la));
+    assertThat(rs.getDoubleArray(10)).isEqualTo(da, 0.0);
+    assertThat(rs.getDoubleArray("da")).isEqualTo(da, 0.0);
+    assertThat(rs.getDoubleList(10)).isEqualTo(Doubles.asList(da));
+    assertThat(rs.getDoubleList("da")).isEqualTo(Doubles.asList(da));
+    assertThat(rs.getBytesList(11)).isEqualTo(Arrays.asList(byta));
+    assertThat(rs.getBytesList("byta")).isEqualTo(Arrays.asList(byta));
+    assertThat(rs.getTimestampList(12)).isEqualTo(Arrays.asList(tsa));
+    assertThat(rs.getTimestampList("tsa")).isEqualTo(Arrays.asList(tsa));
+    assertThat(rs.getDateList(13)).isEqualTo(Arrays.asList(dta));
+    assertThat(rs.getDateList("dta")).isEqualTo(Arrays.asList(dta));
+    assertThat(rs.getStringList(14)).isEqualTo(Arrays.asList(sa));
+    assertThat(rs.getStringList("sa")).isEqualTo(Arrays.asList(sa));
+
     assertThat(rs.next()).isTrue();
     assertThat(rs.getCurrentRowAsStruct()).isEqualTo(struct2);
     assertThat(rs.getString(0)).isEqualTo("y");
     assertThat(rs.getLong(1)).isEqualTo(3L);
     assertThat(rs.isNull(2)).isTrue();
     assertThat(rs.next()).isFalse();
+
+    try {
+      rs.getStats();
+      fail("Exception expected");
```

Comment:
this will fail after my change for DML .. would u mind holding off on ur CL just so that I dont have to redo my PR for DML/PDML.

Reply:
Sure, not a problem. Can you elaborate for my knowledge why it will fail?

Reply:
rs.getStats() returns null instead of an exception ..